renku/core/management/datasets.py

```diff
@@ -128,138 +132,92 @@ def renku_pointers_path(self):
         path.mkdir(exist_ok=True)
         return path
 
-    def datasets_from_commit(self, commit=None):
-        """Return datasets defined in a commit."""
-        commit = commit or self.repo.head.commit
+    @property
+    def datasets(self) -> Dict[str, new_datasets.Dataset]:
+        """A map from datasets name to datasets."""
+        database = Database.from_path(self.database_path)
+        datasets_provenance = DatasetsProvenance(database)
+        return {d.name: d for d in datasets_provenance.datasets}
 
-        try:
-            datasets = commit.tree / self.renku_home / self.DATASETS
-        except KeyError:
-            return
+    # FIXME: Remove this method and use proper injection
+    def get_database(self) -> Database:
```

Comment:
I wanted to get potential injection issues out of the way and focus only on metadata migration since too many tests were failing. This will be addressed in https://github.com/SwissDataScienceCenter/renku-python/issues/2227.